native/src/auth.c

```diff
@@ -10,7 +10,7 @@
 
 AWS_DOTNET_API
 void aws_dotnet_auth_library_init(void) {
-    struct aws_allocator *allocator = aws_dotnet_get_allocator();
```

Comment:
dont you want to pass mem tracer allocator to auth?

Reply:
I'll add some comments.
The idea is prevent the lib init get reported from the memory dump. Since we only clean them up after the dotnet exit. And in tests, we cannot wait until dotnet exit. We are doing similar thing in python:
https://github.com/awslabs/aws-crt-python/blob/main/source/module.c#L977-L978